zookeeper-server/src/main/java/org/apache/zookeeper/server/ZKDatabase.java

```diff
@@ -318,8 +346,14 @@ public void addCommittedProposal(Request request) {
         WriteLock wl = logLock.writeLock();
         try {
             wl.lock();
-            if (committedLog.size() > commitLogCount) {
-                committedLog.remove();
+
+            if (committedLog.size() > commitLogCount || currentCommitLogSize.get() > commitLogSize) {
```

Comment:
The currentCommitLogSize check can't ensure the total payload size. If the incoming committedLog size is getting larger and larger, the currentCommitLogSize will increase continually.